File: examples/sphero-dpad/dpad.go
package main

import (
	"github.com/hybridgroup/gobot"
	"github.com/hybridgroup/gobot/api"
	"github.com/hybridgroup/gobot/platforms/sphero"
	"time"
)

func main() {
	gbot := gobot.NewGobot()
	api := api.NewAPI(gbot)
	api.Port = "8080"
	api.Start()

	adaptor := sphero.NewSpheroAdaptor("Sphero", "/dev/tty.Sphero-YBW-RN-SPP")
	driver := sphero.NewSpheroDriver(adaptor, "sphero")

	robot := gobot.NewRobot("sphero-dpad",
		[]gobot.Connection{adaptor},
		[]gobot.Device{driver},
		func() {},
	)

	robot.AddCommand("move", func(params map[string]interface{}) interface{} {
		direction := params["direction"].(string)

		switch direction {
		case "up":
			driver.Roll(100, 0)
		case "down":
			driver.Roll(100, 180)
		case "left":
			driver.Roll(100, 270)
		case "right":
			driver.Roll(100, 90)
		}

		time.Sleep(2 * time.Second)
		driver.Stop()
		return "ok"
	})

	gbot.AddRobot(robot)

	gbot.Start()
}

File: examples/sphero-dpad/dpad.js
var Cylon = require('cylon');

Cylon.config({
  api: {
    ssl: false,
    port: '8080',
    host: '0.0.0.0',
  }
});

Cylon.api();

Cylon.robot({
  name: 'sphero-dpad',
  connection: { name: 'sphero', adaptor: 'sphero', port: '/dev/tty.Sphero-YBW-RN-SPP' },
  device: { name: 'sphero', driver: 'sphero' },

  move: function(direction) {
    var my = this;

    switch (direction) {
      case "up":
        my.sphero.roll(100, 0);
        break;
      case "down":
        my.sphero.roll(100, 180);
        break;
      case "left":
        my.sphero.roll(100, 270);
        break;
      case "right":
        my.sphero.roll(100, 90);
    }

    setTimeout(function() {
      my.sphero.stop();
    }, 2000);

    return "ok";
  },
  commands: function() {
    return {
      move: this.move
    };
  }
}).start();
